extensions/amp-smartlinks/0.1/linkmate-options.js

```diff
@@ -52,11 +52,9 @@ function getNrtvAccountName_(element) {
  * @private
  */
 function getLinkmateFlag_(element) {
-  const linkmateEnabled = element.getAttribute('linkmate').toLowerCase();
-  userAssert(linkmateEnabled,
-      'amp-smartlinks: linkmate is a required field');
+  const linkmateEnabled = element.getAttribute('linkmate');
 
-  return linkmateEnabled === 'true';
+  return !!linkmateEnabled && linkmateEnabled.toLowerCase() === 'true';
```

Comment:
Given the new validation rule, I believe `linkmateEnabled` can never be a string. 
```
attrs: {
    name: "linkmate"
    value: ""
  }	
```
@honeybadgerdontcare @Gregable to confirm

If so the check will look like `return !!element.hasAttribute('linkmate')`. And you'll need to change your example and test files as well.

Reply:
It can be a string. It can be `"linkmateEnabled"` or empty string.

Reply:
@honeybadgerdontcare can it be arbitrary string? What about
```
<amp-smartlinks linkmate> //empty string
```
```
<amp-smartlinks linkmate="linkmateEnabled"> //arbitrary string
```
```
<amp-smartlinks linkmate="true"> // "true"
```
```
<amp-smartlinks linkmate=true> // boolean
```

Reply:
`value: ""` means the attribute is only valid when it does not have a value or when that value is the same as the attribute name:

```
<amp-smartlinks linkmate></amp-smartlinks> // attribute valid
<amp-smartlinks linkmate=linkmate></amp-smartlinks> // attribute valid
<amp-smartlinks linkmate="true"></amp-smartlinks> // attribute invalid
<amp-smartlinks linkmate=true></amp-smartlinks> // attribute invalid
```

Side note that `=true` and `="true"` are the same thing to the validator, the value is the string `true`.

Reply:
Thank you @honeybadgerdontcare for the clarification! @PhilWinchester looks like `hasAttribute('linkmate')` is the right way to go.